src/Illuminate/Queue/DatabaseQueue.php

```diff
@@ -184,18 +184,24 @@ protected function buildDatabaseRecord($queue, $payload, $availableAt, $attempts
      *
      * @param  string  $queue
      * @return \Illuminate\Contracts\Queue\Job|null
+     * @throws \Exception|\Throwable
      */
     public function pop($queue = null)
     {
         $queue = $this->getQueue($queue);
 
-        $this->database->beginTransaction();
+        try {
+            $this->database->beginTransaction();
 
-        if ($job = $this->getNextAvailableJob($queue)) {
-            return $this->marshalJob($queue, $job);
-        }
+            if ($job = $this->getNextAvailableJob($queue)) {
+                return $this->marshalJob($queue, $job);
```

Comment:
The `return` statement will prevent the `$this->database->commit();` from ever executing. However, there's also a `commit` call inside `marshalJob()`. That could lead to confusion. Can we remove the commit call from `marshalJob` and avoid this early return with a temporarily variable?

Reply:
we discussed removing `commit` with @sisve - he correctly pointed out that might be a breaking change as the `marshalJob()` does not commit the transaction anymore (if any thirdparty code happen to rely on that).

This breaking change could be added to 5.6 branch and I would rather then wrap the whole transaction in the closure to make it cleaner.